pkg/services/services.go

```diff
@@ -136,33 +136,31 @@ func hybridOverlayConfiguration(vxlanPort string, debug bool) servicescm.Service
 
 // kubeProxyConfiguration returns the Service definition for kube-proxy
 func kubeProxyConfiguration(debug bool) servicescm.Service {
-	sanitizedSubnetAnnotation := strings.ReplaceAll(nodeconfig.HybridOverlaySubnet, ".", "\\.")
-	cmd := fmt.Sprintf("%s -log-file=%s %s --windows-service --proxy-mode=kernelspace --feature-gates=WinOverlay=true,WinDSR=true "+
-		"--hostname-override=NODE_NAME --kubeconfig=%s --cluster-cidr=NODE_SUBNET "+
-		"--network-name=%s --source-vip=ENDPOINT_IP --enable-dsr=true", windows.KubeLogRunnerPath, windows.KubeProxyLog,
-		windows.KubeProxyPath, windows.KubeconfigPath, windows.OVNKubeOverlayNetwork)
+	cmd := fmt.Sprintf("%s -log-file=%s %s --source-vip=ENDPOINT_IP --config %s", windows.KubeLogRunnerPath, windows.KubeProxyLog,
+		windows.KubeProxyPath, windows.KubeProxyConfigPath)
 	// Set log level
 	cmd = fmt.Sprintf("%s %s", cmd, klogVerbosityArg(debug))
+	sanitizedSubnetAnnotation := strings.ReplaceAll(nodeconfig.HybridOverlaySubnet, ".", "\\.")
 	return servicescm.Service{
-		Name:    windows.KubeProxyServiceName,
-		Command: cmd,
-		NodeVariablesInCommand: []servicescm.NodeCmdArg{
-			{
-				Name:               "NODE_NAME",
-				NodeObjectJsonPath: "{.metadata.name}",
-			},
-			{
-				Name:               "NODE_SUBNET",
-				NodeObjectJsonPath: fmt.Sprintf("{.metadata.annotations.%s}", sanitizedSubnetAnnotation),
-			},
-		},
+		Name:         windows.KubeProxyServiceName,
+		Command:      cmd,
+		Dependencies: []string{windows.HybridOverlayServiceName},
 		PowershellPreScripts: []servicescm.PowershellPreScript{{
 			VariableName: "ENDPOINT_IP",
-			Path:         windows.NetworkConfScriptPath,
+			Path:         windows.NetworkConfScriptPath + " -NodeName NODE_NAME -NodeSubnet NODE_SUBNET",
+			NodeArgs: []servicescm.NodeCmdArg{
+				{
+					Name:               "NODE_NAME",
+					NodeObjectJsonPath: "{.metadata.name}",
+				},
+				{
+					Name:               "NODE_SUBNET",
+					NodeObjectJsonPath: fmt.Sprintf("{.metadata.annotations.%s}", sanitizedSubnetAnnotation),
+				},
+			},
```

Comment:
How do the node variables in the `Path` get replaced? Might need to have WICD do this before it runs the script

Reply:
do you mean that the NODE_NAME and NODE_SUBNET are variables in the path? For the script, the NodeObjectJsonPath is basically a json gotten from an api call that contains things like the name and annotation.
this sets the args passed to the script to the value gotten from that API call

Reply:
Could you give an example of what WICD actually executes on the Windows node to run the script?

I guess don't see what mechanism is filling in the NODE_NAME and NODE_SUBNET values into
```
network-conf.ps1 -NodeName NODE_NAME -NodeSubnet NODE_SUBNET
```